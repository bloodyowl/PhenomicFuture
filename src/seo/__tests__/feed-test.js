const createRSS = require("../feed")

const posts = [
  {
    title: "a",
    url: "foo",
    description: "bar",
    categories: ["a", "b"],
    author: "bloodyowl",
    date: "2016-12-17",
  },
  // this one should be filtered (too old)
  {
    title: "should not appear",
    url: "foo",
    description: "bar",
    categories: ["a", "b"],
    author: "bloodyowl",
    date: "2015-12-16",
  },
]

describe("createRSS", () => {

  it("should generate a RSS feed", () => {
    return createRSS(posts)
      .then(xml => expect(xml).toMatchSnapshot())
  })

  it("should filter old items in feed", () => {
    return createRSS(posts)
      .then(xml => expect(xml).not.toContain("should not appear"))
  })

})
