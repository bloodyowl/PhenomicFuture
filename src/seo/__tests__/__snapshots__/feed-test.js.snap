exports[`createRSS should generate a RSS feed 1`] = `"<?xml version=\"1.0\" encoding=\"UTF-8\"?><rss xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:content=\"http://purl.org/rss/1.0/modules/content/\" xmlns:atom=\"http://www.w3.org/2005/Atom\" version=\"2.0\"><channel><title><![CDATA[Untitled RSS Feed]]></title><description><![CDATA[Untitled RSS Feed]]></description><link>http://github.com/dylang/node-rss</link><generator>RSS for Node</generator><lastBuildDate>Sat, 17 Dec 2016 14:41:23 GMT</lastBuildDate><atom:link href=\"https://putaindecode.io\" rel=\"self\" type=\"application/rss+xml\"/><item><title><![CDATA[a]]></title><link>https://putaindecode.ioundefined</link><guid isPermaLink=\"true\">https://putaindecode.ioundefined</guid><dc:creator><![CDATA[bloodyowl]]></dc:creator><pubDate>Sat, 17 Dec 2016 00:00:00 GMT</pubDate></item></channel></rss>"`;
